pallets/services-payment/src/lib.rs

```diff
@@ -342,16 +290,54 @@ impl<T: Config> AuthorNotingHook<T::AccountId> for Pallet<T> {
         _block_number: BlockNumber,
         para_id: ParaId,
     ) -> Weight {
-        let total_weight = T::DbWeight::get().reads_writes(1, 1);
+        if Pallet::<T>::burn_credit_for_para(&para_id).is_err() {
```

Comment:
We could rename that function to emphasize that it only burns free credits

```suggestion
        if Pallet::<T>::burn_free_credit_for_para(&para_id).is_err() {
```